### DISCUSSION TITLE
Changing cart and user offcanvas to modal

### DISCUSSION THREAD

**Discussion by SnowYerry** [bootscore/bootscore#595] [Q&A] [answered]
I need help; I haven't been able to find the file to modify the Ajax Cart, which appears as a modal. Similarly, I'm facing the same issue with the profile. If someone could assist me, I would appreciate it.

**Comment by crftwrk** [comment]
> ...modify the Ajax Cart, which appears as a modal

Both, cart and user appears as offcanvas not as modal. Guess that you want to change the offcanvas into a modal, right? If so, you have to edit several files. Note that this is a sensitive area and there will be changes in near future. 

### Cart

#### AJAX

https://github.com/bootscore/bootscore/blob/main/woocommerce/inc/ajax-add-to-cart.php

Create folder in your child and copy file to it:

`woocommerce/inc/ajax-add-to-cart.php`

Override function with new path in your child's functions.php:

```php
function register_ajax_cart() {
  require_once('woocommerce/inc/ajax-add-to-cart.php');
}
add_action('after_setup_theme', 'register_ajax_cart');
```

Now you can edit this file in your child.

#### Offcanvas

https://github.com/bootscore/bootscore/blob/main/template-parts/header/offcanvas-woocommerce.php

Override this file as a usual template file by copying to your child.

#### Offcanvas cart toggler 

https://github.com/bootscore/bootscore/blob/main/template-parts/header/actions-woocommerce.php

Override this file as a usual template file by copying to your child.

### User

Same files as above:

- User offcanvas https://github.com/bootscore/bootscore/blob/main/template-parts/header/offcanvas-woocommerce.php
- User toggler https://github.com/bootscore/bootscore/blob/main/template-parts/header/actions-woocommerce.php

Solved?

**Reply by SnowYerry** [reply]
I didn't explain myself very well, I'm sorry. I don't need to transform them into modals; what I actually need to know is where the information is located and where the login data is entered. I need to add a link similar to the one that says "forgot your password?" but now it should say "create user" and direct to the page I have already created previously.

**Reply by crftwrk** [reply, marked as answer]
Ok, now I understand. User offcanvas content is stored in https://github.com/bootscore/bootscore/blob/main/woocommerce/myaccount/my-account-offcanvas.php.

This isn't a WooCommerce template file and cannot be overridden in the normal way. You have to copy and link it in your child.

1. Copy `my-account-offcanvas.php` to your child using the same folder structure as in main-theme. 
2. Copy https://github.com/bootscore/bootscore/blob/main/template-parts/header/offcanvas-woocommerce.php to your child using the same folder structure as in main-theme.
3. Open `offcanvas-woocommerce.php` and in line `22` you will find this:

https://github.com/bootscore/bootscore/blob/7d7fbac9d0880c32625a864b863457e0d022d736/template-parts/header/offcanvas-woocommerce.php#L22

4. Change `get_template_directory` to `get_stylesheet_directory`:

```php
<?php include get_stylesheet_directory() . '/woocommerce/myaccount/my-account-offcanvas.php'; ?>
```

5. Edit `my-account-offcanvas.php` in your child as you want.

Solved?
